### ISSUE TITLE
Brewing from case classes creates ambiguity

### ISSUE THREAD

**Issue by ftucky** [yakivy/jam#13]
```
trait Config

object defaultConfig extends Config

class Service(cfg:Config)

case class A(some:Config = defaultConfig)

def brew(a:A):Service =
  jam.brewFromRec[Service](a)
```

Compilation fails with :
```
More than one injection candidate was found for Config:
A._1(Config), 
A.some(Config),
A.copy$default$1(Config @uncheckedVariance)
```

The report is perfectly correct.
Being able to use case classes as carriers would be nice though.

**Comment by yakivy** [comment]
@ftucky makes sense to me, I will try to take a look this week

**Comment by yakivy** [comment]
Actually it was a pretty small fix, I've just published it, try `0.4.5`. Feel free to reopen the issue if it doesn't solve your problem.

**Comment by ftucky** [comment]
The fix works fine. Thank you.
